tests/common/dualtor/dual_tor_io.py

```diff
@@ -0,0 +1,424 @@
+
+        # Sender and sniffer have finished the job. Start examining the collected flow
+        self.examine_flow()
+        if self.lost_packets:
+            self.no_routing_stop, self.no_routing_start = datetime.datetime.fromtimestamp(self.no_routing_stop), datetime.datetime.fromtimestamp(self.no_routing_start)
+            logger.error("The longest disruption lasted %.3f seconds. %d packet(s) lost." % (self.max_disrupt_time, self.max_lost_id))
+            logger.error("Total disruptions count is %d. All disruptions lasted %.3f seconds. Total %d packet(s) lost" % \
+                (self.disrupts_count, self.total_disrupt_time, self.total_disrupt_packets))
+
+
+    def generate_from_t1_to_server(self):
+        """
+        Generate (not send) the packets to be sent from T1 to server
+        """
+        eth_dst = self.dut_mac
+        eth_src = self.ptfadapter.dataplane.get_mac(0, 0)
+        ip_dst = DOWNSTREAM_DST_IP
+        ip_ttl = 255
+        tcp_dport = 5000
+
+        if self.tor_port:
+            self.from_tor_src_port = self.tor_port
+        else:
+            self.from_tor_src_port = random.choice(self.port_channel_ports)
+
+        logger.info("-"*20 + "T1 to server packet" + "-"*20)
+        logger.info("Ethernet address: dst: {} src: {}".format(eth_dst, eth_src))
+        logger.info("IP address: dst: {} src: random".format(ip_dst))
+        logger.info("L4 dst: {}".format(tcp_dport))
+        logger.info("DUT mac: {}".format(self.dut_mac))
+        logger.info("-"*50)
+
+        self.packets_list = []
+        for i in xrange(self.packets_to_send):
+            tcp_tx_packet = testutils.simple_tcp_packet(
+                eth_dst=eth_dst,
+                eth_src=eth_src,
+                ip_dst=ip_dst,
+                ip_src=self.random_host_ip(),
+                ip_ttl=ip_ttl,
+                tcp_dport=tcp_dport)
+            payload =  str(i) + 'X' * 60
+            packet = scapyall.Ether(str(tcp_tx_packet))
+            packet.load = payload
+            self.packets_list.append((self.from_tor_src_port, str(packet)))
+
+
+    def generate_from_server_to_t1(self):
+        """
+        Generate (not send) the packets to be sent from server to T1
+        """
+        if self.server_port:
+            self.from_server_src_port = self.server_port
+        else:
+            self.from_server_src_port = random.choice(self.vlan_ports)
+        self.from_server_src_addr  = random.choice(self.vlan_host_map[self.from_server_src_port].keys())
+        self.from_server_dst_addr  = self.random_host_ip()
+        tcp_dport = 5000
+        tcp_tx_packet = testutils.simple_tcp_packet(
+                      eth_dst=self.dut_mac,
+                      ip_src=self.from_server_src_addr,
+                      ip_dst=self.from_server_dst_addr,
+                      tcp_dport=tcp_dport
+                 )
+
+        self.packets_list = []
+        for i in xrange(self.packets_to_send):
```

Comment:
Prefer `range` for Python3 compatibility?

Reply:
Done